src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py

```diff
@@ -0,0 +1,1405 @@
+            >>> model = TFWav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base-960h")
+
+            >>> def map_to_array(batch): >>> speech, _ = sf.read(batch["file"]) >>> batch["speech"] = speech >>> return
+            batch
+
+            >>> ds = load_dataset("patrickvonplaten/librispeech_asr_dummy", "clean", split="validation") >>> ds =
+            ds.map(map_to_array)
+
+            >>> input_values = processor(ds["speech"][0], return_tensors="tf").input_values # Batch size 1 >>> logits =
+            model(input_values).logits >>> predicted_ids = tf.argmax(logits, axis=-1)
+
+            >>> transcription = processor.decode(predicted_ids[0])
+
+            >>> # compute loss >>> target_transcription = "A MAN SAID TO THE UNIVERSE SIR I EXIST"
+
+            >>> # wrap processor as target processor to encode labels >>> with processor.as_target_processor(): >>>
+            labels = processor(transcription, return_tensors="tf").input_ids
+
+            >>> loss = model(input_values, labels=labels).loss
+        """
+
+        inputs = input_processing(
+            func=self.call,
+            config=self.config,
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+            kwargs_call=kwargs,
+        )
+
+        outputs = self.wav2vec2(
+            input_ids=inputs["input_ids"],
+            attention_mask=inputs["attention_mask"],
+            token_type_ids=inputs["token_type_ids"],
+            position_ids=inputs["position_ids"],
+            head_mask=inputs["head_mask"],
+            inputs_embeds=inputs["inputs_embeds"],
+            output_attentions=inputs["output_attentions"],
+            output_hidden_states=inputs["output_hidden_states"],
+            return_dict=inputs["return_dict"],
+            training=inputs["training"],
+        )
+        hidden_states = outputs[0]
+        hidden_states = self.dropout(hidden_states, training=inputs["training"])
+
+        logits = self.lm_head(hidden_states)
+
+        if labels is not None:
+            attention_mask = (
+                attention_mask if attention_mask is not None else tf.ones_like(inputs["input_ids"], dtype=tf.float32)
+            )
+            input_lengths = self._get_feat_extract_output_lengths(tf.sum(attention_mask, axis=-1))
+
+            labels_mask = labels >= 0
+            target_lengths = tf.reduce_sum(labels_mask, axis=-1)
+            flattened_targets = tf.sequence_mask(target_lengths, maxlen=target_lengths)
+
+            log_probs = tf.nn.log_softmax(logits, axis=-1)
+
+            loss = tf.nn.ctc_loss(
```

Comment:
@patrickvonplaten I originally had this part in, along with the `_compute_mask_indices` for SpecAugment. However, I think it makes more sense to do all of the training-specific parts on a separate PR along with `TFWav2Vec2ForPretraing`

Reply:
Think it would be a good idea to leave the ctc_loss in. Wav2Vec2Pretraining is actually a huge PR on its own so I think it would be good to add SpecAugment + `...ForCTC` here and `TFWav2Vec2Pretraining` in a new PR :-)

Reply:
Sounds good. I should be able to knock out SpecAugment and CTC by the end of today. Once this PR is squared away, I will work on the TFWav2Vec2Pretraining one also. I've looked at the in-progress one for Pytorch and I have a couple of questions, but I will save those for that specific PR.